neutronclient/client.py

```diff
@@ -239,6 +248,8 @@ def authenticate(self):
             self._authenticate_keystone()
         elif self.auth_strategy == 'noauth':
             self._authenticate_noauth()
+        elif self.auth_plugin:
+            self._plugin_auth(self.auth_url)
```

Comment:
I'm torn. A plugin, something inherently optional, changing DEFAULT behavior automatically seems like a problem to me. Seems like controlling what the "default" auth strategy is should itself be configurable, else it's Keystone.